pkg/virtctl/portforward/portforward.go

```diff
@@ -0,0 +1,197 @@
+	cmd.Flags().StringVar(&address, addressFlag, address,
+		fmt.Sprintf("--%s=: Set this to the addres the local ports should be opened on", addressFlag))
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type PortForward struct {
+	address      *net.IPAddr
+	clientConfig clientcmd.ClientConfig
+	resource     portforwardableResource
+}
+
+func (o *PortForward) Run(cmd *cobra.Command, args []string) error {
+	kind, namespace, name, ports, err := o.prepareCommand(args)
+	if err != nil {
+		return err
+	}
+
+	if err := o.setResource(kind, namespace); err != nil {
+		return err
+	}
+
+	if forwardToStdout {
+		if len(ports) != 1 {
+			return errors.New("only one port supported when forwarding to stdout")
+		}
+		return o.startStdoutStream(namespace, name, ports[0])
+	}
+
+	o.address, err = net.ResolveIPAddr("", address)
+	if err != nil {
+		return err
+	}
+
+	if err := o.startPortForwards(namespace, name, ports); err != nil {
+		return err
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	<-c
+
+	return nil
+}
+
+func (o *PortForward) prepareCommand(args []string) (kind string, namespace string, name string, ports []forwardedPort, err error) {
+	kind, namespace, name, err = parseTarget(args[0])
+	if err != nil {
+		return
+	}
+
+	ports, err = parsePorts(args[1:])
+	if err != nil {
+		return
+	}
+
+	if len(namespace) > 0 {
+		namespace, _, err = o.clientConfig.Namespace()
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+func (o *PortForward) setResource(kind, namespace string) error {
+	client, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	if kind == "vmi" {
+		o.resource = client.VirtualMachineInstance(namespace)
+	} else if kind == "vm" {
+		o.resource = client.VirtualMachine(namespace)
+	}
+
+	return nil
+}
+
+func (o *PortForward) startStdoutStream(namespace, name string, port forwardedPort) error {
+	streamer, err := o.resource.PortForward(name, port.remote)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(os.Stderr, "forwarding to %s/%s:%d\n", namespace, name, port.remote)
+	if err := streamer.Stream(kubecli.StreamOptions{
+		In:  os.Stdin,
+		Out: os.Stdout,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *PortForward) startPortForwards(namespace, name string, ports []forwardedPort) error {
+	for _, port := range ports {
+		forwarder := portForwarder{
+			namespace: namespace,
+			name:      name,
+			resource:  o.resource,
+		}
+		if err := forwarder.startForwarding(o.address, port); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func usage() string {
+	return `  # Forward the local port 8080 to the vmi port:
+  {{ProgramName}} portforward testvmi 8080
+
+  # Forward the local port 8080 to the vmi port 9090:
+  {{ProgramName}} portforward testvmi 8080:9090
+
+  # Forward the local port 8080 to the vmi port 9090 as a UDP connection:
+  {{ProgramName}} portforward --namespace test testvmi udp/8080:9090
```

Comment:
UDP is not implemented yet, either remove the line and hold of on it or wait until it's done.